src/main/java/com/mixpanel/android/util/RemoteService.java

```diff
@@ -10,7 +10,7 @@
 
 
 public interface RemoteService {
-    boolean isOnline(Context context);
+    boolean isOnline(Context context, OfflineMode offlineMode);
```

Comment:
Hi @juanchospotify ! You have great points and I considered them before writing my comment. I should have explained it better. Would love to hear what you think:

Although this feature is very practical, nobody else has requested it and that's why I wanted to be more hidden (i.e. inside a `RemoteService` implementation and not in the interface itself). Ideally, `RemoteService` assures that `isOnline` is implemented an its implementor would have its own logic. Sadly, we don't let devs give its own `RemoteService` implementation right now and you can only do this on `HttpService` which -I agree- seems a bit odd. Ideally, I would solve this problem by implementing my own `RemoteService` (e.g. `SpotifyHttpService`) that holds a dependency with a feature I want but it's not enforced by the interface.

Since you can't pass your own `RemoteService` implementation the dilemma I had yesterday was whether this should be enforced on the interface level or implement it in `HttpService` and follow up with a new PR that makes having your own `RemoteService` implementation possible.

Personally, I think that it's is a better approach for the good of the project but I am not strong on this based on the current SDK limitations. I won't push it back if you are not happy with it.